src/parcsr_ls/par_ilu_solve_device.c

```diff
@@ -1142,7 +1187,11 @@ hypre_ILUSolveRAPGMRESDevice(hypre_ParCSRMatrix   *A,
          /* perm back */
          if (perm)
          {
+#if defined(HYPRE_USING_SYCL)
+            hypreSycl_scatter(xtemp_data, xtemp_data + n, perm, ftemp_data);
```

Comment:
Not for this PR, we could think of a way to wrap gather/scatter for thrust and oneDPL in a single call

Reply:
Yeah, I think this is part of a much larger question about whether/how we want to unify standard-library-ish calls implemented either in thrust or oneDPL. It might be nice to have our own `hypre_` wrappers for all this functionality, but it also might be too much work to develop/maintain.

Reply:
Good! Seems like a topic for discussion in one of the SLS's